src/writeData/subscriptions/components/CreateSubscriptionPage.tsx

```diff
@@ -80,6 +99,60 @@ const CreateSubscriptionPage: FC = () => {
     getAll<Bucket>(state, ResourceType.Buckets).filter(b => b.type === 'user')
   )
   const {bucket} = useContext(WriteDataDetailsContext)
+  const [completedSteps, setCompletedSteps] = useState<CompletedSteps>(
+    DEFAULT_COMPLETED_STEPS
+  )
+  const [stepsStatus, setStepsStatus] = useState<StepsStatus>(
+    DEFAULT_STEPS_STATUS
+  )
+
+  const getActiveStep = useCallback(activeForm => {
+    let currentStep = 1
+    navigationSteps.forEach((step, index) => {
+      if (step.type === activeForm) {
+        currentStep = index + 1
+      }
+    })
+    return currentStep
+  }, [])
+
+  useEffect(() => {
+    setCompletedSteps({
+      [Steps.BrokerForm]: stepsStatus.brokerStepCompleted === 'true',
+      [Steps.SubscriptionForm]:
+        stepsStatus.subscriptionStepCompleted === 'true',
+      [Steps.ParsingForm]: stepsStatus.parsingStepCompleted === 'true',
+    })
+  }, [stepsStatus])
+
+  useEffect(() => {
+    const status = {
+      currentStep: active,
+      clickedStep: navigationSteps[getActiveStep(active) - 1].type,
+      brokerStepCompleted:
+        formContent.name && formContent.brokerHost && formContent.brokerPort
+          ? 'true'
+          : 'false',
+      subscriptionStepCompleted:
+        formContent.topic &&
+        formContent.bucket &&
+        formContent.bucket !== '<BUCKET>'
+          ? 'true'
+          : 'false',
+      parsingStepCompleted:
+        formContent.dataFormat &&
+        checkRequiredJsonFields(formContent) &&
+        checkRequiredStringFields(formContent)
+          ? 'true'
+          : 'false',
+      dataFormat: formContent.dataFormat ?? 'not chosen yet',
```

Comment:
thoughts on moving this business logic to the `utils/form.ts` file? I think we could add functions like `isBrokerStepCompleted()` and just invoke them here. That way all the logic around form validation is in the same file.

Reply:
i was thinking of moving this to context with the refactoring. but this is a good idea. let me move it.